pkg/traceql/engine_metrics_test.go

```diff
@@ -512,9 +512,10 @@ func TestQuantileOverTime(t *testing.T) {
 		},
 	}
 
-	result, err := runTraceQLMetric(req, in)
+	result, seriesCount, err := runTraceQLMetric(req, in)
 	require.NoError(t, err)
 	require.Equal(t, out, result)
+	assert.Equal(t, len(result), seriesCount)
```

Comment:
Use `require.Equal`?   Actually this might be worth talking about more broadly, but is there ever a reason to use `assert`?  I haven't found a case where continuing a test is helpful after failing and I always prefer `require`.